sessionctx/variable/sysvar.go

```diff
@@ -1075,7 +1075,7 @@ var defaultSysVars = []*SysVar{
 		s.SetEnableCascadesPlanner(TiDBOptOn(val))
 		return nil
 	}},
-	{Scope: ScopeGlobal | ScopeSession, Name: TiDBEnableIndexMerge, Value: Off, Type: TypeBool, SetSession: func(s *SessionVars, val string) error {
+	{Scope: ScopeGlobal | ScopeSession, Name: TiDBEnableIndexMerge, Value: On, Type: TypeBool, SetSession: func(s *SessionVars, val string) error {
```

Comment:
This will only change the default for new installations. I believe this is fine, but I will wait for you to confirm (and update the release note).

If you'd like to change for existing installs, a bootstrap task is needed (session/bootstrap.go).

Reply:
(1) Please make sure this works for 4.0 to 5.0 upgrade too, since from analyze-2.0 feature we found it only works for intra-5.0 upgrade but does not work for the 4.0 to 5.0 upgrade which changes the default anyway no matter it is existing or new installs;
(2) Also @guo-shaoge  is working on GA index merge in Sprint 6. I'd suggest to make this change together with index merge GA feature.

Reply:
I think just change the default for new installations is enough. Change for existing installs will results in plan changes, which will give the user "surprise". That's not what I expected.
And I will talk to @guo-shaoge later.

Reply:
The problem here is that approach we identify whether it's an existing or new install only works for 5.x but not 4.x, which means all the existing system from 4.x will be impacted and default will be changed after upgrading to 5.x.

Reply:
1. #22102 changed the default `oom-action`, which works for all situations(3.0 to 4.0 and 4.0 to 5.0).
But `oom-action` is a config item, I am not sure it works for sysvars.
2. IMO, it's better to enable index merge after it's official GA. We are working on it.